docs/advanced_usage.rst

```diff
@@ -371,3 +371,36 @@ The resulting prompt changes and looks like::
 
 .. note::
    As you can see the order of the options changed from ``1 - MIT`` to ``1 - Apache Software License 2.0``. **Cookiecutter** takes the first value in the list as the default.
+
+.. _`template extensions`:
+
+Template Extensions
+-------------------
+
+*New in Cookiecutter 1.4*
+
+A template may extend the Cookiecutter environment with custom `Jinja2 extensions`_,
+that can add extra filters, tests, globals or even extend the parser.
+
+To do so, a template author must specify the required extensions in ``cookiecutter.json`` as follows:
+
+.. code-block:: json
+
+    {
+        "project_slug": "Foobar",
+        "year": "{% now 'utc', '%Y' %}",
+        "_extensions": ["jinja2_time.TimeExtension"]
+    }
+
+On invocation Cookiecutter tries to import the extensions and add them to its environment respectively.
+
+In the above example, Cookiecutter provides the additional tag `now`_, after
+installing the `jinja2_time.TimeExtension`_ and enabling it in ``cookiecutter.json``.
+
+Please note that Cookiecutter will **not** install any dependencies on its own!
```

Comment:
One of my concerns is that with the need for dependencies beyond the normal ones required by Cookiecutter, we're going to have a lot of 'breaking' templates. Specifically, template authors will add extensions but without a built-in way to mention that they are needed, users will suddenly have problems with templates.

Not in this pull request, but in a future one, I would like to see something like this added to `cookiecutter.json`:

``` JSON
{
   ...
   "_dependencies": ["jinja2_pluralize", "jinja2_timeline"]
}
```